### DISCUSSION TITLE
stack type  exogenous in NBEATSx

### DISCUSSION THREAD

**Discussion by SINAABBASI** [Nixtla/neuralforecast#390] [Q&A] [answered]
Hello. I read the original paper about NBEATSx, and its claimed that there is a new stack specifically to capture the effect of exogenous variates. However, in your current implementation, the stack types for NBEATSx is just `['seasonality', 'trend', 'identity']`, where can I find the 'exogenous' stack type?

**Comment by kdgutier** [comment, marked as answer]
Hi @SINAABBASI,

Thanks for opening this issue, for the moment NeuralForecast's NBEATSx implementation does not have the exogenous projection.

If you want to explore the decomposition with the exogenous projection, it is available in this repository:
https://github.com/cchallu/nbeatsx

As soon as we have spare time we can take a look to the NBEATSx decomposition code.

**Reply by SINAABBASI** [reply]
Thanks a lot for the fast reply. 
Just curious, If its not implemented yet, do you know why when I try nbeatsx with different hist_exgn variates I will get different results?

**Reply by kdgutier** [reply]
For the moment, NBEATSx treating future exogenous and past exogenous variables similarly.

Models like TFT/MQCNN and in our library TCN/LSTM/RNN use an architecture that differentiate the future and past exogenous processing, with a future alignment technique. See this plot: https://github.com/Nixtla/neuralforecast/blob/main/nbs/models.rnn.ipynb

In the NBEATSx paper, the version that had the ability of performing exogenous projections was inspired by these networks.
